rosbag2_transport/src/rosbag2_transport/player.cpp

```diff
@@ -76,6 +77,17 @@ rclcpp::QoS publisher_qos_for_topic(
     topic.name,
     rosbag2_storage::from_rclcpp_qos_vector(topic.offered_qos_profiles));
 }
+
+/// Comparator for SerializedBagMessageSharedPtr to order chronologically by recv_timestamp.
+struct BagMessageRecvTimestampComparator
+{
+  bool operator()(
+    const rosbag2_storage::SerializedBagMessageSharedPtr & l,
+    const rosbag2_storage::SerializedBagMessageSharedPtr & r) const
+  {
+    return l->recv_timestamp > r->recv_timestamp;
+  }
+};
```

Comment:
It would be better to move `BagMessageRecvTimestampComparator` inside `PlayerImpl` class and rename it with a snake style
```suggestion
/// Comparator for SerializedBagMessageSharedPtr to order chronologically by recv_timestamp.
struct
{
  bool operator()(
    const rosbag2_storage::SerializedBagMessageSharedPtr & l,
    const rosbag2_storage::SerializedBagMessageSharedPtr & r) const
  {
    return l->recv_timestamp > r->recv_timestamp;
  }
} BagMessageRecvTimestampComparator;
```

Reply:
I've made those changes in ca23e7fae501e6a2950f529ef0dafa2f37dd1f26.

However, I disagree with changing

```c
struct $name
{
};
```

to

```c
struct
{
} $name;
```

just to be able to change `$name{}` to `$name` when it's used in the constructor, because it doesn't mean the same thing. The first one is a struct type definition, while the second one is a struct type definition (anonymous) + instantiation. It becomes an attribute. Since the goal is to be able to choose a comparator (`recv_timestamp`, `send_timestamp`, or some other comparator), we do not need an instance of each. We just need the struct definitions and then we can instantiate the right one depending on `PlayOptions`, for example.

Reply:
AI describes the second variant as
> struct
 {
 } name;
Anonymous struct:
This syntax defines an anonymous struct. It means the struct doesn't have a name, but you create an instance of it directly with the variable name.
Limited usage:
Anonymous structs are less commonly used. They are useful when you only need a single instance of the struct and don't need to reference it by name elsewhere in your code.

However, this is exactly our use case and the way how we are going to use it. i.e. we need only a single instance of it when passing it to the `message_queue_` constructor.

Reply:
but the `Player` class doesn't need to have an instance of this comparator struct. It's only needed by the priority queue.

Reply:
anyway, as I said above, I've made the change.